### PR TITLE
ci(test): add SSR test framework

### PR DESCRIPTION
**Pull request by jw-foss**
- Setup cypress framework for e2e tests

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: vite.config.cy.mjs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { defineConfig } from 'vite'
```
Comment (sxzz) [inline vite.config.cy.mjs:1]: How about move this file to `/cypress`?

File: e2e/button/button.spec.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { mount } from '@cypress/vue'
```
Comment (sxzz) [inline e2e/button/button.spec.cy.tsx:1]: Why move to top level `/e2e` directory?

Comment (JeremyWuuuuu) [reply e2e/button/button.spec.cy.tsx:1]: If we keep that inside `packages/components`, we will need to update `jest.config` for skipping cypress tests in this case I think we should keep it under `e2e`

Comment (sxzz) [reply e2e/button/button.spec.cy.tsx:1]: I can help you, could you move it to `packages/components` and I will update `tsconfig.json`

Comment (JeremyWuuuuu) [reply e2e/button/button.spec.cy.tsx:1]: It was intended to keep them outside the package. Please just accept it as-is.

**Pull request merged**